newsfragments/4810.added.md

```diff
@@ -0,0 +1 @@
+Optimizes `nth` and `nth_back` for `BoundListIterator`
```

Comment:
maybe also mention that `advance_by` and `advance_by_back` are implemented on nightly